src/Storage/Migration/v0.00/03-setup-functions.sql

```diff
@@ -1,4 +1,151 @@
+            AND (_process_ended_eq  IS NULL OR i.instance -> 'Process' ->> 'Ended' =  _process_ended_eq)
+            AND (_process_isComplete IS NULL OR (_process_isComplete = TRUE AND i.instance -> 'Process' -> 'Ended' IS NOT NULL) OR (_process_isComplete = FALSE AND i.instance -> 'Process' -> 'CurrentTask' IS NOT NULL))
+            AND ((_status_isActiveOrSoftDeleted IS NULL OR _status_isActiveOrSoftDeleted = false) OR ((i.instance -> 'Status' -> 'IsArchived')::boolean = false OR (i.instance -> 'Status' -> 'IsSoftDeleted')::boolean = true))           
+            AND (_status_isArchived IS NULL OR  (i.instance -> 'Status' -> 'IsArchived')::boolean = _status_isArchived)
+            AND ((_status_isArchivedOrSoftDeleted IS NULL OR _status_isArchivedOrSoftDeleted = false) OR ((i.instance -> 'Status' -> 'IsArchived')::boolean = true OR (i.instance -> 'Status' -> 'IsSoftDeleted')::boolean = true))                
+            AND (_status_isHardDeleted IS NULL OR  (i.instance -> 'Status' -> 'IsHardDeleted')::boolean = _status_isHardDeleted)
+            AND (_status_isSoftDeleted IS NULL OR  (i.instance -> 'Status' -> 'IsSoftDeleted')::boolean = _status_isSoftDeleted)                     
+            AND (_visibleAfter_gte IS NULL OR i.instance ->> 'VisibleAfter' >= _visibleAfter_gte)
+            AND (_visibleAfter_gt  IS NULL OR i.instance ->> 'VisibleAfter' >  _visibleAfter_gt)
+            AND (_visibleAfter_lte IS NULL OR i.instance ->> 'VisibleAfter' <= _visibleAfter_lte)
+            AND (_visibleAfter_lt  IS NULL OR i.instance ->> 'VisibleAfter' <  _visibleAfter_lt)
+            AND (_visibleAfter_eq  IS NULL OR i.instance ->> 'VisibleAfter' =  _visibleAfter_eq)
+        ORDER BY
+            (CASE WHEN _sort_ascending = true  THEN i.lastChanged END) ASC,
+            (CASE WHEN _sort_ascending = false THEN i.lastChanged END) DESC,
+            i.id
+        FETCH FIRST _size ROWS ONLY
+    )
+        SELECT filteredInstances.id, filteredInstances.instance, d.element FROM filteredInstances LEFT JOIN storage.dataelements d ON filteredInstances.id = d.instanceInternalId
+        ORDER BY
+            (CASE WHEN _sort_ascending = true  THEN filteredInstances.lastChanged END) ASC,
+            (CASE WHEN _sort_ascending = false THEN filteredInstances.lastChanged END) DESC,
+            filteredInstances.id;
+END;
+$BODY$;
+
+CREATE OR REPLACE FUNCTION storage.readdeletedinstances()
+    RETURNS TABLE (instance JSONB)
+    LANGUAGE 'plpgsql'
+    
+AS $BODY$
+BEGIN
+RETURN QUERY
+    -- Make sure that part of the where clause is exactly as in filtered index instances_isharddeleted_confirmed
+    SELECT i.instance FROM storage.instances i
+    WHERE (i.instance -> 'Status' -> 'IsHardDeleted')::BOOLEAN
+        AND (i.instance -> 'CompleteConfirmations') IS NOT NULL
+        AND (i.instance -> 'Status' ->> 'HardDeleted')::TIMESTAMPTZ <= (NOW() - (7 ||' days')::INTERVAL);
```

Comment:
What is the difference between `->` and `->>`?

Reply:
-> is the native type and ->> casts to text